cumulusci/tasks/salesforce/PublishCommunity.py

```diff
@@ -0,0 +1,60 @@
+from cumulusci.tasks.salesforce import BaseSalesforceApiTask
+from cumulusci.core.exceptions import SalesforceException
+
+
+class PublishCommunity(BaseSalesforceApiTask):
+    api_version = "46.0"
+    task_docs = """
+    Publishes a Salesforce Community via the Connect API.
+    """
+    task_options = {
+        "name": {
+            "description": "The name of the community to publish.",
+            "required": True,
+        },
+        "communityid": {
+            "description": "The id of the community to publish.",
+            "required": False,
+        },
+    }
+
+    def _init_options(self, kwargs):
+        super(PublishCommunity, self)._init_options(kwargs)
+
+    def _run_task(self):
+        community_id = self.options.get("communityid", None)
+        community_name = self.options["name"]
+
+        if community_id is None:
+            self.logger.info(
+                'Finding id for community "{}"'.format(self.options["name"])
+            )
+            community_list = self.sf.restful("connect/communities")["communities"]
+            communities = {c["name"]: c for c in community_list}
+
+            if self.options["name"] in communities:
+                community_id = communities[self.options["name"]]["id"]
+            else:
+                raise SalesforceException(
+                    'Unable to find a community named "{}"'.format(community_id)
+                )
+        else:
+            self.logger.info('Checking name for community "{}"'.format(community_id))
+            community = self.sf.restful("connect/communities/{}".format(community_id))
+            if community_name != community["name"]:
+                raise SalesforceException(
+                    'The community name for {} is "{}" and does not match "{}", the name you provided'.format(
+                        community_id, community["name"], community_name
+                    )
+                )
+
+        self.logger.info(
+            'Sending request to publish Community "{}"({})'.format(
+                community_name, community_id
+            )
+        )
+        response = self.sf.restful(
+            "connect/communities/{}/publish".format(community_id), method="POST"
+        )
+
+        self.logger.info(response["message"])
```

Comment:
What are your thoughts about polling until the publish operation is completed?

Reply:
@davidmreed This is not a polling action, but a one-and-done in which the API simply informs you it has 'started' the process and will email you shortly. So it's an immediate response.

Reply:
Is there any way we can detect when publishing is completed by polling for something? We'll want to use this in flows for setting up an org where we need to know that the community is published before proceeding (to something like running a Robot Framework test that interacts with the community, for example).